docs/reference/triggers/kafka.md

```diff
@@ -1,110 +1,315 @@
+  <br/>
+  **Type:** `int`
+  <br/>
+  **Default Value:** `4`
+  <!-- sarama `Rebalance.Retry.Max` -->
+
+- <a id="rebalanceRetryBackoff"></a>**`rebalanceRetryBackoff`** (**`kafka-rebalance-retry-backoff`**) - Back-off time between retries during rebalancing.
+  <br/>
+  **Type:** `time.Duration` (`string` for annotation)
+  <br/>
+  **Default Value:** `2 * time.Second` (2 seconds)
+  <!-- sarama `Rebalance.Retry.Backoff` -->
+
+- <a id="maxWaitHandlerDuringRebalance"></a>**`maxWaitHandlerDuringRebalance`** (**`kafka-max-wait-handler-during-rebalance`**) - The amount of time to wait for a handler to return when a rebalancing occurs before restarting the worker or replica.
+  <br/>
+  **Type:** `time.Duration` (`string` for annotation)
+  <br/>
+  **Default Value:** `5 * time.Second` (5 seconds)
+
+<a id="rebalancing-config-choice"></a>
 ### Choosing the right configuration for rebalancing
-In a perfect world, Nuclio would be configured out of the box to perform in the most optimal way across all use cases. In fact, if your worst case event processing time is short (a few seconds) then Nuclio does just that - you can leave the defaults as is and Nuclio should perform optimally under normal network conditions. However, if your worst case event processing time is in the order of tens of seconds or minutes, you must choose through configuration between *minimizing duplicate processing* and *higher overall throughput*.
 
-#### I prefer higher throughput (the default)
-Under rebalancing, replicas stop processing while the new generation of the consumer group stabilizes and all members are allocated their partitions. This means that while rebalancing is taking place, messages from Kafka are not processed causing the average pipeline throughput to be reduced. Ideally, once a rebalancing process is initiated due to one of the reasons explained above - all replicas immediately stop what they're doing and join the process. However, when event processing is long, the workers in the replica may be busy processing an event for a while (i.e. the user's handler received an event and has not returned yet). 
+In a perfect world, Nuclio would be configured out of the box to perform in the most optimal way across all use cases. In fact, if your worst-case event-processing time is short (a few seconds), then Nuclio does just that: you can leave the default configurations as-is and Nuclio should perform optimally under normal network conditions. However, if your worst-case event-processing time is in the order of tens of seconds or minutes, you must choose between the following configuration alternatives:
+
+- [Prioritizing throughput (default)](#rebalancing-config-choice-throughput-prioritize)
```

Comment:
Confusing - user must choose between "Prioritizing throughput (default)" and "Prioritizing minimum duplicates". The note is just a note

Reply:
But in the previous structure, the note seemed to be part of the duplicates section, when in fact it applies to both methods, which is also confusing.

Reply:
Granted, but above all it must be clear that there are only two options to choose from. 
I have no idea regarding the note right now.

Reply:
I renamed the "Message pre-fetching note" section to "Rebalancing notes" and moved it from from a subsection of "Choosing the right configuration for rebalancing" to a direct child of the parent "Rebalancing" section — see commit 2db4153.